src/net/ossl.cc

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2015 Cloudius Systems
+ */
+
+#ifdef SEASTAR_MODULE
+module;
+#endif
+
+#include <system_error>
+#include <openssl/err.h>
+
+#ifdef SEASTAR_MODULE
+module seastar;
+#else
+#include <seastar/net/tls.hh>
+#include <seastar/core/sstring.hh>
+#endif
+
+namespace seastar {
+
+class ossl_error_category : public std::error_category {
```

Comment:
so tls.hh exposes this method which i must have implemented:
```
    const std::error_category& error_category();
```
admittedly its probably not as helpful as one would assume since not all errors in OSSL are handled the same way. I try to pass the top error code to the exception when applicable